### ISSUE TITLE
[video_player] Support for Embedded Captions

### ISSUE THREAD

**Issue by mis-kcn** [flutter/flutter#109153]
## Use case

For accessibility, video_player should be able to support Embedded/Closed Captioning, specifically the CEA-608/EIA-608/CTA-708.

## Proposal

At current implementation, video_player does not support or detect CEA-608 captions. Testing out with [ExoPlayer's demo application](https://exoplayer.dev/demo-application.html) at 2.18.0 the captions are getting recognized and toggleable. 

Changing the implementation to use StyledPlayerView works well. An example package would be [video_360](https://pub.dev/packages/video_360) with some modifications to not display in a 360 video format.

**Comment by mis-kcn** [comment]
This is found in Flutter 3.1, video_player 2.4.6. Platform: Android. Haven't tested with iOS yet.
